### ISSUE TITLE
Retraction Wipe, Wipe only when crossing perimeters - dysfunctional

### ISSUE THREAD

**Issue by CuredPrusa** [supermerill/SuperSlicer#4293]
Did I miss something or it's true that checkbox for "Retraction Wipe - Wipe only when crossing perimeters" doesn't work for vast range of SS version, including the latest? 

![image](https://github.com/supermerill/SuperSlicer/assets/98842248/bf1f4b71-92e3-4b93-8512-ba9352d767b6)

I discovered that just recently when I didn't want retraction wipe unless nozzle goes outside the model's wall.

**Comment by CuredPrusa** [comment]
Workaround: Retract only when crossing perimeter =true + Wipe while retracting =true + Retract on layer change =false

That brings similar functionality but disables retractions inside the model. 
Nevertheless, this checkbox looks dysfunctional.

**Comment by supermerill** [comment]
There was a typo, it needs `avoid_crossing_perimeters` to be able to be enabled.
other than that, it works as expected:
with
![image](https://github.com/supermerill/SuperSlicer/assets/6536403/cd299a0d-0467-4004-aec5-b3fdc6f11107)
without
![image](https://github.com/supermerill/SuperSlicer/assets/6536403/7412adc5-bcd4-474c-90dc-b064febfc8fd)

**Comment by CuredPrusa** [comment]
Yeah, it's a bit awkward. 
However, I can not get all retractions inside the model and wipe only when crossing perimeter while crossing perimeter is NOT restricted. That's the whole point of wipe - to make crossing perimeter nicer. If I restrict crossing perimeter, I don't need wipe. If I don't restrict crossing perimeter, I actually need wipe. But I don't want it to wipe at every single retraction. I need wipe only when crossing perimeter, while perimeter crossing is not restricted.
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/f9a6148e-7e70-4e8e-a091-bc1398246f1c)
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/0f78f72a-51a9-4bd4-ae97-035674b3459d)
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/93e2addc-63e1-4274-bc41-9c0d227eff03)
This combination of check boxes gives me this:
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/bab5d6a6-f2fd-44b2-9e87-4867e7b58a9c)
But I want this, with free crossing perimeter:
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/322411a8-7660-4d03-86ee-33ad1a4f50f6)

**Comment by CuredPrusa** [comment]
You can test this on project which I have already sent you (test4) with this modifier:
![image](https://github.com/supermerill/SuperSlicer/assets/98842248/5b75702d-79a0-48f0-b3ca-bf881187aeb5)

**Comment by supermerill** [comment]
understood, i'll check if the restriction can be lifted without too much work.

**Comment by supermerill** [comment]
The issue is that currently, wipe can only be done with a retraction. To decouple the two, a good amount of work & testing is needed.

**Comment by CuredPrusa** [comment]
Well, wipe should be done with a retraction. But retraction should not be done with wipe ;)
I could use a workaround (avoid_crossing_perimeters_max_detour =1mm) .
Is it also difficult to disable wipe at last retraction before travel to seam?
(last screenshot here)

**Comment by supermerill** [comment]
if it's the last travel, maybe there is a layer change after, and if the retraction is enforced for the layer change, there will be a wipe?

**Comment by CuredPrusa** [comment]
Yes, true, retraction is enforced before layer change. But wipe is not necessary if it is nor crossing perimeter. Wipe only when crossing perimeter.
